File: omni-task/src/main/scala/com/awesome/omni/task/Ping.scala
package com.awesome.omni.task

import scalaj.http.Http

class Ping extends Task {
  
  def execute: Unit = {
    
    println("PING")
    
    val response = Http("http://google.com/").asString
    
    println(response.code)
  }

  def report: ExecutionReport = {
    ???
  }

}

File: omni-task/src/main/scala/com/awesome/omni/task/Ping.java
package com.awesome.omni.task;

import java.util.Date;

public class Ping implements Task {

	@Override
	public void execute() {
		System.out.println("ping");
		
	}

	@Override
	public ExecutionReport report() {
		return new ExecutionReport("1", "2", new Date(), new Date(), "ping ping ping");
	}

}
